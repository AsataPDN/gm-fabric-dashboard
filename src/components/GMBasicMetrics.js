import { PropTypes } from "prop-types";
import React from "react";

GMBasicMetrics.propTypes = {
  detailLines: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

export default function GMBasicMetrics({ detailLines, title }) {
  return (
    <div className="kv kv-hero kv-highlight">
      <h3 className="kv-title">
        {title}
      </h3>
      {detailLines.map(([heading, value]) =>
        <div className="kv-pair" key={`${heading}-${value}`}>
          <div className="kv-key">
            {heading}
          </div>
          <div className="kv-value">
            {value}{" "}
          </div>
        </div>
      )}
    </div>
  );
}
