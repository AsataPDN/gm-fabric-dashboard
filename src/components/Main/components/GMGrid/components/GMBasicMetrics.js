import { PropTypes } from "prop-types";
import React from "react";
import {
  Sparklines,
  SparklinesLine,
  SparklinesReferenceLine
} from "react-sparklines";

GMBasicMetrics.propTypes = {
  detailLines: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

/**
 * Basic component for rendering a group of related metrics. Includes sparklines
 * @param {Object} props - refer to propTypes 
 */
export default function GMBasicMetrics({ detailLines, title }) {
  return (
    <div className="kv kv-hero kv-highlight">
      <h3 className="kv-title">{title}</h3>
      {detailLines.map(([heading, value, priority, sparkline = []]) => {
        return (
          <div
            className={`kv-pair kv-pair-${priority}`}
            key={`${heading}-${value}`}
          >
            <div className={`kv-key kv-key-${priority}`}>{heading}</div>
            <div className={`kv-value kv-value-${priority}`}>
              {value.toLocaleString()}
              {sparkline.length > 0 && (
                <div className={`kv-sparkline kv-sparkline-${priority}`}>
                  <Sparklines
                    data={sparkline}
                    height={32}
                    preserveAspectRatio="xMaxYMin"
                  >
                    <SparklinesLine
                      style={{
                        stroke: "currentColor",
                        strokeWidth: 1,
                        fill: "none"
                      }}
                    />
                    <SparklinesReferenceLine
                      style={{
                        stroke: "grey",
                        opacity: "0.4"
                      }}
                      type="mean"
                    />
                  </Sparklines>
                </div>
              )}
            </div>
          </div>
        );
      })}
    </div>
  );
}
