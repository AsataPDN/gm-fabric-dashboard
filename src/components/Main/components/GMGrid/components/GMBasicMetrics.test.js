import React from "react";
import { shallow } from "enzyme";
import GMBasicMetrics from "./GMBasicMetrics";

let wrapper;
const props = {
  detailLines: [
    [
      "2XX",
      3468388,
      "primary",
      [3468388, 3468388, 3468388, 3468388, 3468388, 3468388, 3468388]
    ],
    [
      "200",
      3468388,
      "primary",
      [3468388, 3468388, 3468388, 3468388, 3468388, 3468388, 3468388]
    ],
    ["4XX", 12, "normal", [12, 12, 12, 12, 12, 12, 12]],
    ["400", 8, "normal", [8, 8, 8, 8, 8, 8, 8]],
    ["499", 4, "normal", [4, 4, 4, 4, 4, 4, 4, 4]],
    ["5XX", 16, "normal", [16, 16, 16, 16, 16, 16, 16]],
    ["500", 16, "normal", [16, 16, 16, 16, 16, 16, 16]]
  ],
  title: "Response Status Codes"
};
describe("<GMBasicMetrics> stateless child component of GMGrid", () => {
  beforeEach(() => {
    wrapper = shallow(<GMBasicMetrics {...props} />);
  });

  test("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
