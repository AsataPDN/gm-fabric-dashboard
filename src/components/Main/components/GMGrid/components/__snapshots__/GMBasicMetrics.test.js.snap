// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GMBasicMetrics> stateless child component of GMGrid matches snapshot 1`] = `
<styled.div>
  <styled.h3>
    Response Status Codes
  </styled.h3>
  <styled.div
    key="2XX-3468388"
    priority="primary"
  >
    <styled.div>
      2XX
    </styled.div>
    <styled.div>
      3,468,388
      <styled.div>
        <Sparklines
          data={
            Array [
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="200-3468388"
    priority="primary"
  >
    <styled.div>
      200
    </styled.div>
    <styled.div>
      3,468,388
      <styled.div>
        <Sparklines
          data={
            Array [
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
              3468388,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="4XX-12"
    priority="normal"
  >
    <styled.div>
      4XX
    </styled.div>
    <styled.div>
      12
      <styled.div>
        <Sparklines
          data={
            Array [
              12,
              12,
              12,
              12,
              12,
              12,
              12,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="400-8"
    priority="normal"
  >
    <styled.div>
      400
    </styled.div>
    <styled.div>
      8
      <styled.div>
        <Sparklines
          data={
            Array [
              8,
              8,
              8,
              8,
              8,
              8,
              8,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="499-4"
    priority="normal"
  >
    <styled.div>
      499
    </styled.div>
    <styled.div>
      4
      <styled.div>
        <Sparklines
          data={
            Array [
              4,
              4,
              4,
              4,
              4,
              4,
              4,
              4,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="5XX-16"
    priority="normal"
  >
    <styled.div>
      5XX
    </styled.div>
    <styled.div>
      16
      <styled.div>
        <Sparklines
          data={
            Array [
              16,
              16,
              16,
              16,
              16,
              16,
              16,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="500-16"
    priority="normal"
  >
    <styled.div>
      500
    </styled.div>
    <styled.div>
      16
      <styled.div>
        <Sparklines
          data={
            Array [
              16,
              16,
              16,
              16,
              16,
              16,
              16,
            ]
          }
          height={32}
          margin={2}
          preserveAspectRatio="xMaxYMin"
          width={240}
        >
          <SparklinesLine
            onMouseMove={[Function]}
            style={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeWidth": 1,
              }
            }
          />
          <SparklinesReferenceLine
            style={
              Object {
                "opacity": "0.4",
                "stroke": "grey",
              }
            }
            type="mean"
          />
        </Sparklines>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
